Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx

```diff
@@ -261,11 +298,89 @@ export const TriggerCreationModal: React.FC<TriggerCreationModalProps> = (props)
       });
       return;
     }
-    const newDialog = generateNewDialog(dialogs, dialogId, formData, schemas.sdk?.content);
-    if (formData.$kind === intentTypeKey && !isRegEx) {
+
+    if (formData.$kind === intentTypeKey && isLUISnQnA && formData.triggerPhrases) {
+      const newDialog = generateNewDialog(dialogs, dialogId, formData, schemas.sdk?.content, {});
       const newIntent = { Name: formData.intent, Body: formData.triggerPhrases };
       onSubmit(newDialog, newIntent);
+    } else if (formData.$kind === qnaMatcherKey || formData.$kind === onChooseIntentKey) {
+      const lgTemplateId1 = generateDesignerId();
+      const lgTemplateId2 = generateDesignerId();
+      const extraTriggerAttributes = {};
+      if (formData.$kind === onChooseIntentKey) {
+        extraTriggerAttributes['actions[4].prompt'] = `\${TextInput_Prompt_${lgTemplateId1}()}`;
+        extraTriggerAttributes['actions[5].elseActions[0].activity'] = `\${SendActivity_${lgTemplateId2}()}`;
+        const lgTemplates1: LgTemplate[] = [
+          {
+            name: `TextInput_Prompt_${lgTemplateId1}`,
+            body: '[Activity\n\
+    Attachments = ${json(AdaptiveCardJson())}\n\
+]\n',
+          } as LgTemplate,
+          {
+            name: `SendActivity_${lgTemplateId2}`,
+            body: '- Sure, no worries.',
+          } as LgTemplate,
+        ];
+
+        let lgTemplates2: LgTemplate[] = [
+          {
+            name: 'AdaptiveCardJson',
+            body: adaptiveCardJsonBody,
+          } as LgTemplate,
+          {
+            name: `whichOneDidYouMean`,
+            body: whichOneDidYouMeanBody,
+          } as LgTemplate,
+          {
+            name: 'pickOne',
+            body: pickOne,
+          } as LgTemplate,
+          {
+            name: 'getAnswerReadBack',
+            body: getAnswerReadBack,
+          } as LgTemplate,
+          {
+            name: 'getIntentReadBack',
+            body: getIntentReadBack,
+          } as LgTemplate,
+        ];
+
+        lgTemplates2 = lgTemplates2.filter(
+          (t) => commonlgFile?.templates.findIndex((clft) => clft.name === t.name) === -1
+        );
+
+        const lgFilePayload = {
+          [`${dialogId}.${locale}`]: lgTemplates1,
+          [`common.${locale}`]: lgTemplates2,
+        };
+        const newDialog = generateNewDialog(dialogs, dialogId, formData, schemas.sdk?.content, extraTriggerAttributes);
```

Comment:
instead of use a pre-defined `designerId`, It's better to get `designerId` from the created `newDialog`, then you do not need change a lot of places